pep-0XXX-environment-markers.rst

```diff
@@ -0,0 +1,174 @@
+PEP: 0XXX
+Title: Environment Markers
+Version: $Revision$
+Last-Modified: $Date$
+Author: James Polley <jp@jamezpolley.com>
+Status: Draft
+Type: Informational
+Content-Type: text/x-rst
+Created: 03-Jul-2015
+Post-History: 03-Jul-2015
+Replaces: 0345 (part of)
+
+
+Abstract
+========
+
+An **environment marker** describes a condition about the current execution
+environment. They are used to indicate when certain dependencies are only
+required in particular environments, and to indicate supported platforms
+for distributions with additional constraints beyond the availability of a
+Python runtime.
+
+Environment markers were first specified in PEP-0345[1]. PEP-0426[2] (which
+would replace PEP-0345) proposed extensions to the markers. When
+2.7.10 was released, even these extensions became insufficient due to
+their reliance on simple lexical comparisons, and thus this PEP has
+been born.
+
+
+Rationale
+=========
+
+Many Python packages are written with portability in mind.
+
+For many packages this means they aim to support a wide range of
+Python releases. If they depend on libraries such as argparse or
+simplejson - which started as external libraries, but later got
+incorporated into core - specifying a single set of requirements is
+difficult, as the set of required packages differs depending on the
+version of Python in use.
+
+For other packages, this means broad cross-platform support. However,
+this may mean (for instance) that certain packages are required for
+installation on Windows, but not other platforms.
+
+Environment Markers attempt to provide more flexibility in a list of
+requirements by allowing the developer to give some information about
+the conditions under which a requirement should be installed.
+
+
+Examples
+========
+
+Here are some examples of such markers inside a requirements.txt::
+
+   "sys_platform == 'win32'"
+   "platform_machine == 'i386'"
+   "python_version == '2.4' or python_version == '2.5'"
+   "python_version>='2.7' and python_version<'3.0'"
```

Comment:
This example is aimed at demonstrating environment marker boolean logic, so a better example might be:

    backports.ssl_match_hostname >= 3.4; python_version < '2.7.9' or (python_version >= '3.0' and python_version < '3.4')